### PR TITLE
Updates for v3.25.0

### PR DESCRIPTION
**Pull request by mgleung**
## Description

Updates for v3.25.0

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]

File: calico/_includes/release-notes/v3.25.0-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+##### Windows
+ - Update Windows NSSM version [calico #6861](https://github.com/projectcalico/calico/pull/6861) (@song-jiang)
+ - windows: ensure calico-managed kubelet starts after the calico network has been initialized [calico #6656](https://github.com/projectcalico/calico/pull/6656) (@vitaliy-leschenko)
+
+##### OpenStack
+ - Calico for OpenStack: remove iptables programming by the DHCP agent that is no longer needed, and that was increasing the need for Felix to resync Calico's iptables programming. Existing users will see issues - i.e. a VM failing to learn its IP address at boot time - if their VM OS is old enough to have unfixed DHCP client software.  In that case the remedy is to update the VM OS.  For example, in Tigera's own testing, we updated from CirrOS 0.3.4 to CirrOS 0.6.0. [calico #6857](https://github.com/projectcalico/calico/pull/6857) (@tj90241)
+ - Calico for OpenStack: prime the project (aka tenant) data cache on Neutron server startup [calico #6839](https://github.com/projectcalico/calico/pull/6839) (@tj90241)
+ - Allow Calico to set MTU in OpenStack [calico #6725](https://github.com/projectcalico/calico/pull/6725) (@nelljerram)
+
```
Comment (huoqifeng) [inline calico/_includes/release-notes/v3.25.0-release-notes.md:106]: Shall we mention s390x support in 3.25.0 https://github.com/projectcalico/calico/pull/5793?

File: apiserver/Makefile [hunk 2]
```diff
@@ -55,6 +55,11 @@ VERSION_FLAGS = -X $(PACKAGE_NAME)/cmd/apiserver/server.VERSION=$(APISERVER_VERS
 
 include ../lib.Makefile
 
+# TODO Add s390x back to VALIDARCHES. Excluding it should be temporary since
+# there are currently issues in our pipeline with building s390x images. Remove
+# this command once the s390x build pipeline has been fixed.
+VALIDARCHES=amd64 arm64 ppc64le
```
Comment (huoqifeng) [inline apiserver/Makefile:61]: s390x support has been added in https://github.com/projectcalico/calico/pull/5793, https://github.com/projectcalico/calico/blob/release-v3.25/apiserver/docker-image/Dockerfile.s390x

**Review by huoqifeng — COMMENTED** [review]
Shall we mention s390x support in 3.25 @caseydavenport @mgleung ?

**Comment by caseydavenport** [comment]
@huoqifeng I believe @mgleung had trouble getting s390x to build properly for the release. He can explain in more detail what issues he was encountering.

**Comment by mgleung** [comment]
@huoqifeng there were some build issues with s390x for the apiserver so I ended up removing that build for v3.25.0 so it would not block the release. You can see the exact issues in the test run for https://github.com/projectcalico/go-build/pull/413 .

On an unrelated note, it looks like our script did not pick up your changes for a release note since a release note was not provided. We can definitely fix that with v3.25.1, but we'll also need to figure out the containerized build issues for s390x.

File: apiserver/Makefile (hunk 2 above)
Comment (mgleung) [reply apiserver/Makefile:61]: Unfortunately there were some build issues with our containerized builds that need to be addressed before the s390x artifacts can be built properly for release. See the tests in https://github.com/projectcalico/go-build/pull/413 for more details.

**Comment by huoqifeng** [comment]
Yeah, I'll be looking at https://github.com/projectcalico/go-build/pull/413

**Pull request merged**